flagsmith-engine/utils/hashing/index.ts

```diff
@@ -41,8 +17,8 @@ function h2d(s: any): string {
 export function getHashedPercentateForObjIds(objectIds: Array<any>, iterations = 1): number {
     let toHash = makeRepeated(objectIds, iterations).join(',');
     const hashedValue = md5(toHash);
-    const hashedInt = bigInt(h2d(hashedValue));
-    const value = (hashedInt.mod(9999).toJSNumber() / 9998) * 100;
+    const hashedInt = BigInt('0x' + hashedValue);
+    const value = (Number((hashedInt % 9999n)) / 9998.0) * 100;
```

Comment:
Will this result in different percentage values being served for users if they upgrade the node client to this version?

Reply:
Both of these need to happen for the values to be exactly the same:

* `bigInt(h2d(hashedValue))` and `BigInt('0x...')` must be equivalent. I think it's fair to assume that `h2d` is correct and does the same conversion as `BigInt('0x...')`. 
* `bigInt` behaves the same way as native `BigInt`. [BigInteger.js](https://github.com/peterolson/BigInteger.js) acts as a polyfill for the native `BigInt` if it's available, so if anyone was running the old in newer Node versions, they were basically running this new code already.

We can't exhaustively test all values (2^128 possible MD5 hashes) but we could write some sampling tests if we want to get more confidence.

Reply:
I'd be happy if we just ran a small sample (~3) of identities through each version of the code to confirm.

Reply:
https://github.com/Flagsmith/flagsmith-nodejs-client/actions/runs/11724377165/job/32658169904?pr=164#step:6:33

From https://github.com/Flagsmith/flagsmith-nodejs-client/pull/164

Reply:
Lovely stuff, thanks.